modules/dynamic/src/smithy4s/dynamic/internals/DynamicModelCompiler.scala

```diff
@@ -289,14 +313,34 @@ private[dynamic] object Compiler {
 
       val valueList = values.map(_._2).toList.sortBy(_.intValue)
 
-      val theEnum = intEnumeration(values.apply, valueList)
+      if (shape.traits.contains(IdRef("alloy#openEnum"))) {
+        val theEnum = enumeration[Int](
+          v =>
+            values.getOrElse(
+              v,
+              EnumValue(
+                stringValue = v.toString,
+                intValue = v,
+                value = v,
+                name = "$Unknown",
+                hints = Hints.empty
+              )
+            ),
+          EnumTag.OpenIntEnum(identity),
+          valueList
+        )
+
+        update(id, shape.traits, theEnum)
+      } else {
+        val theEnum = intEnumeration(values.apply, valueList)
 
-      updateWithHints(
-        id, {
-          allHints(shape.traits)
-        },
-        theEnum
-      )
+        updateWithHints(
+          id, {
+            allHints(shape.traits)
+          },
+          theEnum
+        )
+      }
     }
 
```

Comment:
**issue:** pending - support for 1.x enums. I'll add those later on  in this PR.

Reply:
added now